File: src/ManaFoodProduct.Application/UseCases/ProductUseCase/Commands/CreateProduct/CreateProductCommand.cs
using MediatR;
using ManaFoodProduct.Application.Dtos;
using System.Collections.Generic;

namespace ManaFoodProduct.Application.UseCases.ProductUseCase.Commands.CreateProduct
{
    public record CreateProductCommand(string Name, double UnitPrice, Guid CategoryId, List<Guid>? ItemIds = null) : IRequest<ProductDto>;
}

File: src/main/kotlin/br/com/manafood/manafoodproduct/application/usecase/product/commands/create/CreateProductCommand.kt
package br.com.manafood.manafoodproduct.application.usecase.product.commands.create

import java.math.BigDecimal
import java.util.*

data class CreateProductCommand(
    val name: String,
    val description: String?,
    val unitPrice: BigDecimal,
    val categoryId: UUID,
    val itemIds: List<UUID> = emptyList(),
    val createdBy: UUID
)